### PR TITLE
Improve test clustering and execution

### LINKED ISSUE #113: Extend internal pipeline for unit and integration testing
### Describe the feature request

As a plugin maintainer I want to run all my tests and collect the results.

### Upstream changes

* [x] Jenkinsfile.internal

### Acceptance criteria

* [x] add stage for unit tests
* [x] add stage for integration tests
* [x] upload all test results to test.guide

### LINKED ISSUE #117: Improve test clustering and execution
### Describe the feature request

As a plugin maintainer I want to clearly distinguish my test types and execute them with specific gradle tasks.

### Upstream changes

* [x] build.gradle
 * [x] src/test/groovy

### Acceptance criteria

* [x] add gradle task for unit tests
* [x] add gradle task for integration tests
* [x] outsource integration test cases into *IT.groovy files

### PR DESCRIPTION
**Pull request by MarioFuchsTT**
Resolves #113
Resolves #117

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -210,6 +210,27 @@ test {
     systemProperties('spock.skip.sandbox': System.getProperty('spock.skip.sandbox', 'true'))
 }
```
Comment (ErikRehmTT) [inline build.gradle:211]: just let test depend the sub tasks e.g. 
```
test {
    dependsOn(unitTest, integrationTest)
}
```

so there is no duplication of definitions

**Review by ErikRehmTT — CHANGES_REQUESTED** [review]
please add the updates of #113 here as well

File: build.gradle [hunk 2]
```diff
@@ -210,6 +210,27 @@ test {
...
+        excludeTestsMatching '*ContainerTest'
+        excludeTestsMatching '*IT'
+    }
+    useJUnitPlatform()
+
+    // skip some process tests for ci
+    systemProperties('spock.skip.sandbox': System.getProperty('spock.skip.sandbox', 'true'))
+}
+
+task integrationTest(type: Test) {
```
[the remark this answers is not in the document]
Comment (MarioFuchsTT) [reply build.gradle:220]: [d8e31fb](https://github.com/jenkinsci/ecu-test-execution-plugin/pull/118/commits/d8e31fbccc38836a7f79bd29155a290e66e8f93d)

File: build.gradle (hunk 1 above)
Comment (MarioFuchsTT) [reply build.gradle:207]: resolved in [d8e31fb](https://github.com/jenkinsci/ecu-test-execution-plugin/pull/118/commits/d8e31fbccc38836a7f79bd29155a290e66e8f93d)

**Comment by MarioFuchsTT** [comment]
> please add the updates of #113 here as well

resolved in [59b620b](https://github.com/jenkinsci/ecu-test-execution-plugin/pull/118/commits/59b620bde20148f2306e6ca077fe8b0a678e30fc)

File: build.gradle [hunk 3]
```diff
@@ -195,19 +195,15 @@ tasks.named('server').configure {
...
 test {
-    // exclude code, which can not be instrumented correctly
-    jacoco {
-        excludes = ["com/gargoylesoftware/**"]
-    }
     // exclude tests which have own test tasks
     filter {
         excludeTestsMatching '*ContainerTest'
     }
     useJUnitPlatform()
```
Comment (ErikRehmTT) [inline build.gradle]: I think this would run the tests again. -> remove

Comment (MarioFuchsTT) [reply build.gradle]: resolved in [48699f1](https://github.com/jenkinsci/ecu-test-execution-plugin/pull/118/commits/48699f12ef7ee4419425dece936414be7ee4dc2b)

**Review by ErikRehmTT — APPROVED** [review]

**Pull request merged**